csharp/ql/src/semmle/code/csharp/Type.qll

```diff
@@ -55,27 +55,8 @@ private predicate isObjectClass(Class c) { c instanceof ObjectType }
  * Either a value type (`ValueType`) or a reference type (`RefType`).
  */
 class ValueOrRefType extends DotNet::ValueOrRefType, Type, Attributable, @value_or_ref_type {
-  /** Gets the name of this type without `<...>` brackets, in case it is a constructed type. */
-  private string getNameWithoutBrackets() {
-    exists(UnboundGenericType unbound, string name |
-      unbound = this.(ConstructedType).getUnboundDeclaration() and
-      name = unbound.getName() and
-      result = name.prefix(name.length() - unbound.getNumberOfTypeParameters() - 1)
-    )
-    or
-    not this instanceof ConstructedType and
-    result = this.getName()
-  }
-
-  language[monotonicAggregates]
-  private string getQualifiedTypeArguments() {
-    result =
-      strictconcat(Type t, int i |
-        t = this.(ConstructedType).getTypeArgument(i)
-      |
-        t.getQualifiedName(), "," order by i
-      )
-  }
+  /** Gets the name of this type without `<...>` brackets, in case it is a generic type. */
+  string getNameWithoutBrackets() { types(this, _, result) }
```

Comment:
I suggest we deprecate `getNameWithoutBrackets`, and replace the body of `getUndecoratedName` with `types(this, _, result)`.